pipeline-scripts/buildlib.groovy

```diff
@@ -1337,4 +1337,138 @@ def getChanges(yamlData) {
+                            break
+                        case 'fast':
+                            pr_messages << "Please merge as soon as ${errata_url} is shipped live OR if a Cincinnati-first release is approved."
+                            break
+                        case 'stable':
+                            pr_messages << "Please merge within 48 hours of ${errata_url} shipping live OR a Cincinnati-first release."
+                            break
+                        default:
+                            error("Unknown prefix: ${prefix}")
+                    }
+
+                    if ( addToUpgradeChannel ) {
+                        pr_messages << "This PR will also enable upgrades from ${releaseName} to releases in ${upgradeChannel}"
+                    }
+
+                    if ( !addToChannel && !addToUpgradeChannel ) {
+                        def pr_info = "No Cincinnati PRs opened for ${prefix}. Might have been done by previous architecture's release build.\n"
+                        echo pr_info
+                        currentBuild.description += pr_info
+                        continue
+                    }
+
+                    withCredentials([string(credentialsId: 'openshift-bot-token', variable: 'access_token')]) {
+                        def messageArgs = ''
+                        for ( String msg : pr_messages ) {
+                            messageArgs += "--message '${msg}' "
+                        }
+
+                        sh """
+                            set -euo pipefail
+                            set -o xtrace
+                            if git checkout origin/${branchName} ; then
+                                echo "The branch ${branchName} already exists in cincinnati-graph-data. No additional PRs will be created."
+                                exit 0
+                            fi
+                            git branch -f ${branchName} origin/master
+                            git checkout ${branchName}
+                            if [[ "${addToChannel}" == "true" ]]; then
+                                echo >> ${channelFile}    # add newline
+                                echo '# Errata: ${errata_url}' >> ${channelFile}    # add link to errata for reference
+                                echo '- ${releaseName}' >> ${channelFile}   # add the entry
+                                git add ${channelFile}
+                            fi
+                            if [[ "${addToUpgradeChannel}" == "true" ]]; then
+                                echo >> ${upgradeChannelFile}    # add newline
+                                echo '# Errata: ${errata_url}' >> ${upgradeChannelFile}    # add link to errata for reference
+                                echo '- ${releaseName}' >> ${upgradeChannelFile}   # add the entry
```

Comment:
This is going to add it at the end of the file. It's perhaps a minor thing but if you look at e.g. https://github.com/openshift/cincinnati-graph-data/blob/master/channels/candidate-4.3.yaml the 4.2 "upgrade from" stream is separate from the 4.3 stream. OTA team likes this to be human-readable. Perhaps we could `sed` the line in before the first {major}.{minor}.0 line?

Reply:
The other problem that's evident in that file is that upgrades from 4.2 to 4.3 require the `+amd64` prefix for as long as 4.3 isn't shipping Z. No idea how to handle that.

Reply:
@sosiouxme https://github.com/openshift/doozer/pull/197 for consideration on how to accomplish this.

Reply:
The new push does not use https://github.com/openshift/doozer/pull/197 . Instead, it checks for the existence of the next minor's latest release in the various api.ci namespaces to determine if that release was built / released for each architecture.

Reply:
Current push contains this latter strategy.